src/core/filesystem.cc

```diff
@@ -1042,7 +1044,7 @@ GetFileSystem(const std::string& path, FileSystem** file_system)
         "-DTRTIS_ENABLE_S3=ON.");
 #else
     re2::RE2 s3_regex(
-        "s3://([a-z]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-zA-Z.-_]+)*)?)");
+        "s3://([0-9a-zA-Z-.]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-zA-Z.-_]+)*)?)");
     std::string host_name, host_port, bucket, object;
     if (!RE2::FullMatch(
             path, s3_regex, &host_name, &host_port, &bucket, &object)) {
```

Comment:
Why is CheckClient below commented out?

Reply:
This is residual code from the initial s3 work. The CheckClient was not implemented for s3, only for GCS.

Reply:
Removed